_source/_posts/2023-07-27-spring-graphql-react.md

```diff
@@ -0,0 +1,1794 @@
+    private List<Person> controlledBy = new ArrayList<>();
+
+    public Company(String SIC, String category, String companyNumber, String countryOfOrigin, LocalDate incorporationDate, Integer mortgagesOutstanding, String name, String status) {
+        this.id = null;
+        this.SIC = SIC;
+        this.category = category;
+        this.companyNumber = companyNumber;
+        this.countryOfOrigin = countryOfOrigin;
+        this.incorporationDate = incorporationDate;
+        this.mortgagesOutstanding = mortgagesOutstanding;
+        this.name = name;
+        this.status = status;
+    }
+
+    public Company withId(Long id) {
+        if (this.id.equals(id)) {
+            return this;
+        } else {
+            Company newObject = new Company(this.SIC, this.category, this.companyNumber, this.countryOfOrigin, this.incorporationDate, this.mortgagesOutstanding, this.name, this.status);
+            newObject.id = id;
+            return newObject;
+        }
+    }
+
+    public String getSIC() {
+        return SIC;
+    }
+
+    public void setSIC(String SIC) {
+        this.SIC = SIC;
+    }
+
+    public String getCategory() {
+        return category;
+    }
+
+    public void setCategory(String category) {
+        this.category = category;
+    }
+
+    public String getCompanyNumber() {
+        return companyNumber;
+    }
+
+    public void setCompanyNumber(String companyNumber) {
+        this.companyNumber = companyNumber;
+    }
+
+    public String getCountryOfOrigin() {
+        return countryOfOrigin;
+    }
+
+    public void setCountryOfOrigin(String countryOfOrigin) {
+        this.countryOfOrigin = countryOfOrigin;
+    }
+
+    public LocalDate getIncorporationDate() {
+        return incorporationDate;
+    }
+
+    public void setIncorporationDate(LocalDate incorporationDate) {
+        this.incorporationDate = incorporationDate;
+    }
+
+    public Integer getMortgagesOutstanding() {
+        return mortgagesOutstanding;
+    }
+
+    public void setMortgagesOutstanding(Integer mortgagesOutstanding) {
+        this.mortgagesOutstanding = mortgagesOutstanding;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getStatus() {
+        return status;
+    }
+
+    public void setStatus(String status) {
+        this.status = status;
+    }
+}
+```
+
+Create the package `com.okta.developer.demo.repository` and the class `CompanyRepository`:
+
+__CompanyRepository.java__
+```java
+package com.okta.developer.demo.repository;
+
+import com.okta.developer.demo.domain.Company;
+import org.springframework.data.neo4j.repository.ReactiveNeo4jRepository;
+
+public interface CompanyRepository extends ReactiveNeo4jRepository<Company, Long> {
+
+}
+```
+
+Create the configuration class `GraphQLConfig` under the root package. This class will enable CORS from the React client and log the GraphQL schema mappings:
```

Comment:
Should this say "This class will (eventually) enable CORS)" since it doesn't do it yet?

Reply:
I moved the cors config to properties and added the comment there.